Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -147,8 +149,8 @@ namespace Babylon
         NativeXr();
         ~NativeXr();
 
-        void BeginSession(Napi::Env env); // TODO: Make this asynchronous.
-        void EndSession();                // TODO: Make this asynchronous.
+        arcana::task<void, std::exception_ptr> BeginSession(Napi::Env env, const JsRuntimeScheduler& runtimeScheduler);
```

Comment:
You don't need to do this in this PR, but it would be good to clean up the code for this. BeginSession should probably be renamed to BeginSessionAsync and EndSession should probably become async as well. Also, these functions should actually do the work asynchronously instead of a while loop.

Reply:
I'll look at that after this PR.  I briefly looked at the init code in OpenXR which is the reason why this is a busy while loop (on the Android implementation it just returns true, so breaks on the first pass), but since I was primarily working on Android and this change was already getting big I deferred that change.